src/psyclone/psyir/symbols/datasymbol.py

```diff
@@ -353,7 +366,8 @@ def copy(self):
 
     def copy_properties(self, symbol_in):
         '''Replace all properties in this object with the properties from
-        symbol_in, apart from the name which is immutable.
+        symbol_in, apart from the name and scope (public/private) which are
```

Comment:
I think it makes sense not to copy the scope since that is determined by the declaration of the current symbol, rather than the one in the Container. I've therefore updated the docstring as you suggest.